orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -173,21 +183,24 @@ def forward(ctx, *inputs, **kwargs):
 
                     # TODO: Try to reuse the output buffers as some of the output tensors are same sizes,
                     #   especially the backward graph outputs.
-                    training_io_binding = self._training_session.io_binding()
+                    training_forward_io_binding = self._training_session.io_binding()
+                    training_backward_io_binding = self._training_session.io_binding()
```

Comment:
training_backward_io_binding can be created at backward(), since it's not needed in forward() code?

Reply:
acctually, does io_binding still need to be a part of RunStateInfo? 

RunStateInfo was intended for object sharing between forward and backward, but since forward and backward has their own iobinding now, do we still need to cache the iobinding? 

---
In reply to: [602564659](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r602564659) [](ancestors = 602564659)

Reply:
I created here because RunStateInfo was being created here and there wasn't any other function to pass this other than its constructor. If we can remove IOBindings from RunStateInfo then it would be the best.

---
In reply to: [602565648](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r602565648) [](ancestors = 602565648,602564659)